### PR TITLE
Todo Pow (G_FloatString.cs)

### PR DESCRIPTION
**Pull request by Marsgames**
Replace Pow function with a better algorithm

### CODE REVIEW

File: Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs [hunk 1]
```diff
@@ -207,14 +206,18 @@ public static float ToFloat(this int i)
...
+        // source : https://en.wikipedia.org/wiki/Exponentiation_by_squaring
+        private static int Pow (int f, int p) {
+            if (0 == p) {
+                return 1;
+            } else if (1 == p) {
+                return f;
+            } else if (0 == p % 2) {
+                return Pow (f * f, p / 2);
+            } else {
+                return f * Pow (f * f, (p - 1) / 2);
```
Comment (Therzok) [inline Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: This is not tail recursive though.

Comment (QFSW) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: More importantly why do we have this Pow function and should we just remove it instead?

Comment (br3aker) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: What's the point of implementing a new algorithm if this Pow is called exactly once at initialization stage?

Comment (Marsgames) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: There was a todo and I don't like todoes

Comment (Therzok) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: `Mathf.pow` should be a good alternative in this case.

Comment (br3aker) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: Wait, there was an for-loop implementation? `Mathf.Pow` is better for sure - it's probably using hardware `pow` implementation which is pseudo-"1 tick" implementation, not a loop one. Even if it's not using it, it's still faster/equal to log2 approach.

Comment (Marsgames) [reply Assets/Tayx/Graphy - Ultimate Stats Monitor/Scripts/Util/G_FloatString.cs]: Yes, there was a for loop. I don't really know why Mathf.Pow wasn't use, so I just improved the existing function, knowing it won't be the best approach.

**Comment by Tayx94** [comment]
Thanks all for the discussion 😄 
I've swapped the Pow function with Mathf.Pow.

https://github.com/Tayx94/graphy/commit/db6fd1fabe91b5bc82952bc1c4acdc0afdc95085

**Pull request closed without merging**